cdc/sink/schema_registry.go

```diff
@@ -0,0 +1,259 @@
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Warn("Failed to query the registry",
+			zap.String("uri", uri))
+		return nil, 0, errors.Annotate(err, "Failed to query the registry")
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, 0, errors.Annotate(err, "Failed to read response from Registry")
+	}
+
+	if resp.StatusCode != 200 && resp.StatusCode != 404 {
+		log.Warn("Failed to query schema from the Registry, HTTP error",
+			zap.Int("status", resp.StatusCode),
+			zap.String("uri", uri),
+
+			zap.ByteString("responseBody", body))
+		return nil, 0, errors.New("Failed to query schema from the Registry, HTTP error")
+	}
+
+	if resp.StatusCode == 404 {
+		log.Warn("Specified schema not found in Registry",
+			zap.String("key", key),
+			zap.Int64("tiSchemaID", tiSchemaID))
+
+		return nil, 0, errors.New("Schema not found in Registry")
+	}
+
+	var jsonResp lookupResponse
+	err = json.Unmarshal(body, &jsonResp)
+	if err != nil {
+		return nil, 0, errors.Annotate(err, "Failed to parse result from Registry")
+	}
+
+	cacheEntry := new(schemaCacheEntry)
+	cacheEntry.codec, err = goavro.NewCodec(jsonResp.Schema)
+	if err != nil {
+		return nil, 0, errors.Annotate(err, "Creating Avro codec failed")
+	}
+	cacheEntry.registryID = jsonResp.RegistryID
+	cacheEntry.tiSchemaID = tiSchemaID
+	m.cache[tableNameToSchemaSubject(tableName)] = cacheEntry
+
+	log.Info("Avro schema lookup successful with cache miss",
+		zap.Int64("tiSchemaID", cacheEntry.tiSchemaID),
+		zap.Int("registryID", cacheEntry.registryID),
+		zap.String("schema", cacheEntry.codec.Schema()))
+
+	return cacheEntry.codec, cacheEntry.registryID, nil
+}
+
+// ClearRegistry clears the Registry subject for the given table. Should be idempotent.
+// Exported for testing.
+func (m *AvroSchemaManager) ClearRegistry(tableName model.TableName) error {
+	uri := m.registryURL + "/subjects/" + url.QueryEscape(tableNameToSchemaSubject(tableName))
+	req, err := http.NewRequest("DELETE", uri, nil)
+	if err != nil {
+		log.Error("Could not construct request for clearRegistry", zap.String("uri", uri))
+		return err
+	}
+	req.Header.Add("Accept", "application/vnd.schemaregistry.v1+json, application/vnd.schemaregistry+json, application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Error("Could not send delete request to clear Registry")
+		return err
+	}
+
+	if resp.StatusCode == 200 {
+		log.Info("Clearing Registry successful")
+		return nil
+	}
+	if resp.StatusCode == 404 {
+		log.Info("Clearing Registry: topic does not exists, no-op", zap.String("uri", uri))
+		return nil
+	}
+
+	log.Error("Other error when clearing Registry")
+	return err
+}
+
+func tableNameToSchemaSubject(tableName model.TableName) string {
+	// We should guarantee unique names for subjects
+	return url.QueryEscape(tableName.Schema) + "%" + url.QueryEscape(tableName.Table)
```

Comment:
`%` could be in schema name or table, then both schema `t%`, table `t` and schema `t`, table `%t` will generate the same subject `t%%t`.
Recommend to use 
`model.QuoteSchema(schema, table)` before query escape

Reply:
I use QueryEscape first which replaces a plain "%" with "%25", thus avoiding the conflict. I will change tomodel.QuoteSchema() anyways.